forem/forem
# Replace collective_noun with community_name

## What type of PR is this? (check all applicable)


## Description
This PR removes `collective_noun` and `collective_noun_disabled` from the SiteConfig and any relevant places in the codebase and replaces `collective_noun` with `community_name`. One concern that was voiced on the issue associated with this work was:
> the only issue would be handling instances in which folks didn't want to include the "collective noun" value. For example, when we say "Welcome to DEV" versus "Welcome to the DEV Community"

I realize what I'm about to say is a bit radical, _but_ to resolve this issue, I have replaced `community_qualified_name` with `community_name` and have updated the copy above the "Community Name" textbox in the SiteConfig to be more explicit about what an Admin could put there, e.g. "DEV" or "DEV Community" or even "DEV Flock." In doing this, I have:

- Cut down on some of our code and complexity in the `application_helper.rb`, which keeps our use of `community_name` a little more consistent (there were places in the codebase where we were using `"<%= community_name %> Community"` and in others, `<%= community_qualified_name %>`)

- Given Admins more flexibility around what their "collective noun" is while also generalizing "community name" a bit more, without adding bloat to the SiteConfig

- Removed an edge case where Admins could potentially set their "Community Name" to something along the lines of "DEV Flock" in the SiteConfig, causing our already appended "Community" to turn their "Community Name" in some areas to "DEV Flock Community" without the Admin realizing


## Related Tickets & Documents
Closes https://github.com/forem/forem/issues/11456
Relates to PR #11973 

## QA Instructions, Screenshots, Recordings
To QA this PR, first ensure that you have `admin` and `config` privileges and then please do the following:

- Navigate to `/admin/config` and make sure everything still looks correct now that the `collective_noun` textbox and `collective_noun_disabled` checkbox have been removed from the "Community" section of the config:
![Screen Shot 2020-12-10 at 8 04 47 AM](https://user-images.githubusercontent.com/32834804/101789313-63f30400-3abe-11eb-892b-490dc67995cd.png)

- While testing this out locally, please also look through the CoC, Terms of Use, Contacts page, and any other page where `collective_noun` and/or `community_qualified_name` was replaced to ensure that the pages are still readable:
![Screen Shot 2020-12-10 at 8 09 18 AM](https://user-images.githubusercontent.com/32834804/101789861-03b09200-3abf-11eb-9cd5-8f9e7c42cc77.png)

- Ensure that all of the checks run and are green! ♻️ 

### UI accessibility concerns?
N/A

## Added tests?

      have not been included_
- [x] I removed and cleaned up existing tests

## Added to documentation?

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [x] I have removed the "Collective Noun" section from the Admin Guide and have added some further details to the "Community Name" section, but have not yet merged the changes. To view these changes and suggest edits, please take a look at my proposed changes: https://app.gitbook.com/@forem/s/forem-admin-guide/~/drafts/-MOr2aA_SCj61eSjxPLG/admin/config/community-content

## [optional] Are there any post-deployment tasks we need to perform?
I have included a `data_update` script in this PR that will need to run to update the `community_name` for Forems that do not have `collective_noun_disabled` to append their `collective_noun` to their `community_name`.

## [optional] What gif best describes this PR or how it makes you feel?

![Happy Bill Nye at a computer](https://media.giphy.com/media/l3nWqD4ViFej9REAw/giphy.gif)
